IPython/core/magics/pylab.py

```diff
@@ -84,9 +86,15 @@ def matplotlib(self, line=''):
 
             In [3]: %matplotlib qt
         """
+
         args = magic_arguments.parse_argstring(self.matplotlib, line)
-        gui, backend = self.shell.enable_matplotlib(args.gui)
-        self._show_matplotlib_backend(args.gui, backend)
+
+        if args.list is not None:
```

Comment:
My guess is as as you do `store_true ` above `args.list` can only be `True̵̵̵|False` so the `... is None` is alway True, and you get into the if.